metadatastore/commands.py

```diff
@@ -39,167 +99,512 @@ def inner(*args, **kwargs):
+    leaking out to user code.
+
+    When we migrate to using uid as the primary key this function
+    will be removed.
+
+    Parameters
+    ----------
+    oid : ObjectId
+        Mongo's unique identifier for the document.  This is currently
+        used to implement foreign keys
+
+    Returns
+    -------
+    runstart : doc.Document
+        The RunStart document.
+    """
+    try:
+        return _RUNSTART_CACHE_OID[oid]
+    except KeyError:
+        pass
+    rs = RunStart._get_collection().find_one({'_id': oid})
+    return _cache_runstart(rs)
+
+
+@_ensure_connection
+def _descriptor_given_oid(oid):
+    """Get EventDescriptor document given an ObjectId
+
+    This is an internal function as ObjectIds should not be
+    leaking out to user code.
+
+    When we migrate to using uid as the primary key this function
+    will be removed.
+
+    Parameters
+    ----------
+    oid : ObjectId
+        Mongo's unique identifier for the document.  This is currently
+        used to implement foreign keys
+
+    Returns
+    -------
+    descriptor : doc.Document
+        The RunStart document.
+    """
+    try:
+        return _EVENTDESC_CACHE_OID[oid]
+    except KeyError:
+        pass
+
+    descriptor = EventDescriptor._get_collection().find_one({'_id': oid})
+    return _cache_descriptor(descriptor)
+
+
+@_ensure_connection
+def runstart_given_uid(uid):
+    """Given a uid, return the RunStart document
+
+    Parameters
+    ----------
+    uid : str
+        The uid
+
+    Returns
+    -------
+    runstart : doc.Document
+        The RunStart document.
+
+    """
+    try:
+        oid = _RUNSTART_UID_to_OID_MAP[uid]
+        return _RUNSTART_CACHE_OID[oid]
+    except KeyError:
+        pass
+    runstart = RunStart._get_collection().find_one({'uid': uid})
+    return _cache_runstart(runstart)
+
+
+@_ensure_connection
+def runstop_given_uid(uid):
+    """Given a uid, return the RunStop document
+
+    Parameters
+    ----------
+    uid : str
+        The uid
+
+    Returns
+    -------
+    runstart : doc.Document
+        The RunStop document fully de-referenced
+
+    """
+    try:
+        oid = _RUNSTOP_UID_to_OID_MAP[uid]
+        return _RUNSTOP_CACHE_OID[oid]
+    except KeyError:
+        pass
+    # get the raw runstop
+    runstop = RunStop._get_collection().find_one({'uid': uid})
+    return _cache_runstop(runstop)
+
+
+@_ensure_connection
+def descriptor_given_uid(uid):
+    """Given a uid, return the EventDescriptor document
+
+    Parameters
+    ----------
+    uid : str
+        The uid
+
+    Returns
+    -------
+    runstart : doc.Document
+        The EventDescriptor document fully de-referenced
+
+    """
+    try:
+        oid = _EVENTDESC_UID_to_OID_MAP[uid]
+        return _EVENTDESC_CACHE_OID[oid]
+    except KeyError:
+        pass
+
+    descriptor = EventDescriptor._get_collection().find_one({'uid': uid})
+    return _cache_descriptor(descriptor)
+
+
+def runstop_by_runstart(runstart):
```

Comment:
No, because the `runstart_given_uid` call (that I just added back) is decorated as such, but will add back just to be safe.